core/dnsserver/onstartup.go

```diff
@@ -2,9 +2,16 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
 )
 
+// checkDomainName() returns true if the given domain name follows RFC1035 preferred syntax.
+func checkDomainName(domainName string) bool {
+	matchDomain, _ := regexp.Compile("^[A-Za-z0-9-]{1,}\\.[A-Za-z0-9-]{1,}\\.$")
```

Comment:
This will fail to match the root domain, `.`. 

Can you share the Corefile you are trying with?

Reply:
okay.. let me update Regex for root domain too.